go-controller/pkg/ovn/routeimport/route_import.go

```diff
@@ -343,7 +323,27 @@ func (c *controller) syncNetwork(network string) error {
 	if info == nil {
 		return nil
 	}
-	router := info.GetNetworkScopedGWRouterName(c.node)
+
+	// get the table from the network VRF. Note we go to netlink for this as
```

Comment:
why not try `c.tables` before getting it from netlink?

Reply:
Because it is a cache after all, and the information might be old, so not as trustable. This became more of a concern when we can handle VRFs with any name, and not just UDN specifically named VRFs that nobody is supposed to tinker with. Most likely using `c.tables` is fine, but why not take the safe approach and have less concerns unless we know we can't?

Reply:
I am not against using it, it is just an extra netlink call.
I don't think that the table is just a hint though, in most cases it determines whether we call syncNetwork at all so we already rely on it heavily.

Reply:
Discussed offline but clarifying. 
Yeah, with hint meaning it just used to hint what network to reconcile, which can be spuriously wrong but of no consequence because syncNetwork does not rely on that cache.
Staleness of a table<->network association is temporary until we receive a link update or network update.

I didn't want to give the impression of taking the decision to add an extra netlink call lightly. The more I think about it, the more I would use netlink transactions if they existed. Even inclined to add an extra netlink call just before we push the route ops but that's maybe taking it too far.